extensions/omni_httpc/omni_httpc.c

```diff
@@ -250,12 +261,25 @@ static int on_body(h2o_httpclient_t *client, const char *errstr, h2o_header_t *t
   // End of stream, complete the request
   if (errstr == h2o_httpclient_error_is_eos) {
     SET_VARSIZE(req->body.data, req->body.len);
-    (*req->done)++;
+    req->complete = true;
   }
 
   return 0;
 }
 
+void copy_request_fields(struct request *src, struct request *dest) {
```

Comment:
This function copies pointers, not the actual values. Do you think it's safe and sufficient for this case?

Reply:
according to my limited understanding:
since the request structs are are allocated using `palloc0`, they should live until the current Postgres memory context lives. The memory context seems to live until all of the requests have finished. Therefore it seems to me that copying the pointers should be fine in this case.

Reply:
according to my limited understanding:
since the request structs are are allocated using `palloc0` or `palloc`, they should live until the current Postgres memory context lives. The memory context seems to live until all of the requests have finished. Therefore it seems to me that copying the pointers should be fine in this case. Everything appears to be "automatically" freed here:
https://github.com/omnigres/omnigres/blob/ff598c2be91f5d309cc06939134a35fc8fa79230/extensions/omni_httpc/omni_httpc.c#L624-L623